File: reverse_integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        reverse = 0
        neg_flag = False
        if x < 0:
            neg_flag = True
            x *= -1
        while x!=0:
            dig = x % 10
            reverse = reverse * 10 + dig
            x = x / 10
        if neg_flag == True:
            reverse *= -1
        if reverse < -2147483648 or reverse > 2147483647:
            return 0
        return reverse
        

class Solution(object):
    def reverse(self, x):
        s = cmp(x, 0)
        r = int(`s*x`[::-1])
        return s*r * (r < 2**31)

File: reverse_integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long rev = 0;
        bool neg_flag = false;
        if(x < 0) {
            neg_flag = true;
            x *= -1;
        }   
        while(x!=0) {
            int dig = x % 10;
            rev = rev * 10 + dig;
            x = x / 10;
        }
        if(neg_flag == true) {
            rev *= -1;
        }
            
        if(rev < -2147483648 || rev > 2147483647) {
            return 0;
        }
        return rev;
    }
};


class Solution {
public:
    int reverse(int x) {
        long long res = 0;
        while(x) {
            res = res*10 + x%10;
            x /= 10;
        }
        return (res<INT_MIN || res>INT_MAX) ? 0 : res;
    }
};


// checks overflow
class Solution {
public:
    int reverse(int x) {
        int ans = 0;
        while (x) {
            int temp = ans * 10 + x % 10;
            if (temp / 10 != ans)
                return 0;
            ans = temp;
            x /= 10;
        }
        return ans;
    }
};